sql-plugin/src/main/scala/com/nvidia/spark/rapids/GpuParquetScan.scala

```diff
@@ -114,6 +114,25 @@ object GpuParquetScanBase {
     tagSupport(scan.sparkSession, schema, scanMeta)
   }
 
+  def throwIfNeeded(
+      table: Table,
+      isCorrectedInt96Rebase: Boolean,
+      isCorrectedDateTimeRebase: Boolean): Unit = {
+    (0 until table.getNumberOfColumns).foreach { i =>
+      val col = table.getColumn(i)
+      // if col is a day
+      if (!isCorrectedDateTimeRebase && RebaseHelper.isDateRebaseNeeded(col)) {
+        throw DataSourceUtils.newRebaseExceptionInRead("Parquet")
+      }
+      // if col is a time
+      else if (!isCorrectedInt96Rebase || !isCorrectedDateTimeRebase) {
```

Comment:
This logic looks incorrect.  If INT96 config is specified and says don't throw but the legacy datetime config says throw, we still shouldn't throw on a timestamp column **if** it's INT96.  I think this method needs to take two boolean parameters, one to indicate whether we're checking for a rebase on dates and another that says whether we're checking for a rebase on timestamps.  The caller is responsible for passing the proper boolean, as this code is not checking whether we're writing or reading INT96 here.

I'm also not seeing how the config settings are factoring into this code path.  It looks like the parameters passed are simply what's read from the file, regardless of the read or write config settings specified by the user.